### ISSUE TITLE
Custom Annotations/Labels for different sport 

### ISSUE THREAD

**Issue by Wilann** [SilvioGiancola/SoccerNetv2-DevKit#23]
Hi SoccerNet Dev team again, apologies for the 3rd issue in a row, but I'm really fascinated with this work! 

As a seasoned Badminton fan, I would love to apply this architecture to Badminton matches. Is it possible to do so? 
As I've gathered from the material, there are a few places where I need to change the code to accomplish this: 

1. When using the annotation tool (when I get it working after my issue #21 is solved), I need to change some code so that when I press ENTER the menu should display my custom labels. 

2. In `./Task1-ActionSpotting/CALF/src/config/classes.py`, with custom labels, `EVENT_DICTIONARY_V2` and `INVERSE_EVENT_DICTIONARY_V2` must be changed (to my custom labels, and just incrementing the dictionary value I assume), but does `K_V2` have to change? If so, what does `K_V2` change to, or how do I calculate it? 

3. In `./Task1-ActionSpotting/CALF/src/model.py`, I of course need to make sure the matrices line-up, as I have a different number of class predictions. 

Is there anywhere else the code needs to changed that I'm missing? I haven't yet looked at the code thoroughly for points 1 & 3, and I'm unsure about the theory in point 2, but if anyone could point me in the right direction, that'd be greatly appreciated! 
Thank you so much again for reading, and looking forward to your response!

**Comment by SilvioGiancola** [comment]
Hi @Wilann ,

We appreciate your enthusiasm about SoccerNet, I hope you will be able to apply that on Badminton videos, our work should transfer well to any other sport. Have a look at the other papers published in CVSports, some of them already focuses Badminton. 

To answer your questions:

1. You should be able to adapt the annotation tool once your issue is solved (it's just a library misplaced on your end). You'll have to use your own label of course, it shouldn't be too complicated to update the code.

2. Accordingly, you will have to change/update the `EVENT_DICTIONARY_V2` and `INVERSE_EVENT_DICTIONARY_V2` to match your classes. You will have to update the `K_V2` to reflect the amount of context you want to embed around each class of action. Check the CALF paper, we explain what those values correspond to. AFAIR, we used a bayesian optimization for those hyper-parameters, but the performances were not much sensitive to those values, so I recommend you to start with your own values and eventually fine-tune later.

3. Similarly, you will have to update the classifier in `model.py` to output the same number of classes you have defined in your new dataset. Also, feel free to try out all the methods we provide for action spotting in this devkit.

Let us know if you have any more doubt, Cheers!

**Comment by Wilann** [comment]
Hi @SilvioGiancola, thank you for confirming, and letting me know about CVSports! (I had never heard of it before) 

Regarding my questions:

1. I will come back to this once I get my annotation issue resolved. 

2. `EVENT_DICTIONARY_V2` and `INVERSE_EVENT_DICTIONARY_V2` should be simple enough to change. Thank you for confirming the purpose of `K_V2` - will fine-tune as needed. 

3. Will definitely check out the other methods after I get this one working!

**Comment by Wilann** [comment]
Update: Annotation issue resolved. Will re-open issue if needed.
